package/kedro_viz/server.py

```diff
@@ -317,90 +323,141 @@ def _construct_layer_mapping(catalog):
+def _pretty_name(name):
+    name = name.replace("-", " ").replace("_", " ")
+    parts = [n[0].upper() + n[1:] for n in name.split()]
+    return " ".join(parts)
+
+
+def format_pipelines_data(pipelines, catalog) -> Dict[str, list]:
     """
-    Format pipeline and catalog data from Kedro for kedro-viz
+    Format pipelines and catalog data from Kedro for kedro-viz.
 
     Args:
-        pipeline: Kedro pipeline object
+        pipelines: Dictionary of Kedro pipeline objects.
         catalog:  Kedro catalog object
+
+    Returns:
+        Dictionary of pipelines, nodes, edges, tags and layers list.
+
     """
+    all_pipelines = []
+
+    all_nodes = {}
+    # keep track of node_id -> set(child_node_ids) for layers sorting
+    node_dependencies = defaultdict(set)
+    all_edges = []
+    sorted_nodes = []
+    all_tags = set()
+
+    for pipeline_key, pipeline in pipelines.items():
+        all_pipelines.append({"id": pipeline_key, "name": _pretty_name(pipeline_key)})
+        formated_nodes, edges = format_pipeline_data(
+            pipeline_key,
+            pipeline,
+            catalog,
+            all_nodes,
+            node_dependencies,
+            all_tags,
+            all_edges,
+        )
+        sorted_nodes += formated_nodes
+        all_edges += edges
+
+    # sort tags
+    sorted_tags = [{"id": tag, "name": _pretty_name(tag)} for tag in sorted(all_tags)]
+    # sort layers
+
+    sorted_layers = _sort_layers(all_nodes, node_dependencies)
+
+    return {
+        "nodes": sorted_nodes,
+        "edges": all_edges,
+        "tags": sorted_tags,
+        "layers": sorted_layers,
+        "pipelines": all_pipelines,
+    }
 
-    def pretty_name(name):
-        name = name.replace("-", " ").replace("_", " ")
-        parts = [n[0].upper() + n[1:] for n in name.split()]
-        return " ".join(parts)
+
+# pylint: disable=too-many-locals,too-many-arguments
+def format_pipeline_data(
+    pipeline_key, pipeline, catalog, all_nodes, node_dependencies, all_tags, all_edges
+):
+    """
+    Format pipeline and catalog data from Kedro for kedro-viz
+    Args:
+        pipeline: Kedro pipeline object
+        catalog:  Kedro catalog object
+    """
 
     # keep tracking of node_id -> node data in the graph
-    nodes = {}
+    # nodes = {}
     # keep track of a sorted list of nodes to returned to the client
     sorted_nodes = []
-    # keep track of node_id -> set(child_node_ids) for layers sorting
-    node_dependencies = defaultdict(set)
+
     # keep track of edges in the graph: [{source_node_id -> target_node_id}]
     edges = []
     # keep_track of {data_set_namespace -> set(tags)}
     namespace_tags = defaultdict(set)
     # keep track of {data_set_namespace -> layer it belongs to}
     namespace_to_layer = {}
-    all_tags = set()
 
     dataset_to_layer = _construct_layer_mapping(catalog)
 
     for node in sorted(pipeline.nodes, key=lambda n: n.name):
         task_id = _hash(str(node))
```

Comment:
Here, `str(node)` could be `'func1([bob_in,parameters]) -> [bob_out]'`, so it's relying on the combination of node func/input/output, thus the below case is generating two `func1` dictionalies.